### PR TITLE
[VL] support array_join

### PR DESCRIPTION
**Pull request by Ma-Jian1**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by Ma-Jian1** [comment]
@PHILO-HE Could you review this PR？

**Review by PHILO-HE — APPROVED** [review]
LGTM.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -658,6 +658,14 @@ object ExpressionConverter extends SQLConfHelper with Logging {
           substraitExprName.get,
           replaceWithExpressionTransformer(arrayMin.child, attributeSeq),
           arrayMin)
+      case arrayJoin: ArrayJoin =>
```
[the remark this answers is not in the document]
Comment (PHILO-HE) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:661]: @Ma-Jian1, please have a further check. Thanks!

File: backends-clickhouse/src/main/scala/io/glutenproject/utils/CHExpressionUtil.scala [hunk 2]
```diff
@@ -95,6 +95,7 @@ object CHExpressionUtil {
   )
 
   final val CH_BLACKLIST_SCALAR_FUNCTION: Map[String, FunctionValidator] = Map(
+    ARRAY_JOIN -> DefaultValidator(),
```
Comment (PHILO-HE) [inline backends-clickhouse/src/main/scala/io/glutenproject/utils/CHExpressionUtil.scala:98]: @exmy, @zzcclp, please note this change for CH.

Comment (exmy) [reply backends-clickhouse/src/main/scala/io/glutenproject/utils/CHExpressionUtil.scala:98]: It's ok.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ma-Jian1) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:661]: Ok, I'll double check it.

[the remark this answers is not in the document]
Comment (Ma-Jian1) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala]: It's covered by `normalFunctionExpressionTransform`, revert it.

**Review by PHILO-HE — APPROVED** [review]
LGTM. Assume it is covered by the imported spark UT.

**Pull request merged**